src/ansys/units/dimensions.py

```diff
@@ -1,43 +1,12 @@
-"""Provides the ``Dimensions`` and ``BaseDimensions`` class."""
-from enum import Enum
-from typing import Optional
+"""Provides the ``Dimensions`` class."""
+from typing import Union
 
-
-class BaseDimensions(Enum):
-    """
-    Supplies all valid base dimensions used in dimensional analysis.
-
-    Used as dictionary keys for defining a `Dimensions` object.
-
-    Attributes
-    ----------
-    MASS
-    LENGTH
-    TIME
-    TEMPERATURE
-    TEMPERATURE_DIFFERENCE
-    ANGLE
-    CHEMICAL_AMOUNT
-    LIGHT
-    CURRENT
-    SOLID_ANGLE
-    """
-
-    MASS = 0
-    LENGTH = 1
-    TIME = 2
-    TEMPERATURE = 3
-    TEMPERATURE_DIFFERENCE = 4
-    ANGLE = 5
-    CHEMICAL_AMOUNT = 6
-    LIGHT = 7
-    CURRENT = 8
-    SOLID_ANGLE = 9
+import ansys.units as ansunits
 
 
 class Dimensions:
     """
-    A class which contains the base unit information.
+    Tracks the power of all base unit types for dimensional analysis.
```

Comment:
```suggestion
    A composite dimension (or simply dimensions) composed from an arbitrary number of fundamental dimensions, where each fundamental dimension is a pair consisting of base dimension and exponent.
```

Reply:
N.b. we are use power in a few place where we could say exponent. This is important because power has a physical connotation.